File: authentication/python/post.py
import time
import json
import hmac
import hashlib
import base64
import requests
import warnings
from dotenv import load_dotenv
import os
from os.path import join, dirname

warnings.filterwarnings("ignore")

load_dotenv()

if os.getenv('API_KEY') is None or os.getenv('SECRET_KEY') is None:
    raise ValueError("Required environment variables (API_KEY, SECRET_KEY) are missing.")

api_key = os.getenv('API_KEY')
secret_key = os.getenv('SECRET_KEY')

payload = {
    'method': 'POST',
    'path': '/v4/orders',
    'body': {
         'amount': 0.01,
         'pair': 'ABC_DEF',
         'price': 300000, 
         'side': 'buy', 
         'type': 'limit'
    }
}

timestamp = int(time.time() * 1000)
body_string = json.dumps(payload['body'], separators=(',', ':'))
message = f"{timestamp}{payload['method']}{payload['path']}{body_string}"
signature = hmac.new(secret_key.encode(), message.encode(), hashlib.sha256).digest()
signature_base64 = base64.b64encode(signature).decode()

url = f"https://api.ripiotrade.co{payload['path']}"

headers = {
    'Content-Type': 'application/json',
    'Authorization': api_key,
    'Signature': signature_base64,
    'Timestamp': str(timestamp)
}

response = requests.post(url, headers=headers, data=body_string, verify=False)

print('response:', response.json())

File: authentication/php/post.php
<?php
// Load environment variables from .env file
require_once __DIR__ . '/vendor/autoload.php';

$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->load();

// Check if required environment variables are set
if (!isset($_ENV['API_KEY']) || !isset($_ENV['SECRET_KEY'])) {
    throw new Exception("Required environment variables (API_KEY, SECRET_KEY) are missing.");
}

$apiKey = $_ENV['API_KEY'];
$secretKey = $_ENV['SECRET_KEY'];

$payload = [
    'method' => 'POST',
    'path' => '/trade/orders',
    'body' => [
        'amount' => 0.01,
        'pair' => 'ABC_DEF',
        'price' => 300000,
        'side' => 'buy',
        'type' => 'limit'
    ]
];

$method = $payload['method'];
$path = parse_url($payload['path'], PHP_URL_PATH);
$body = json_encode($payload['body'], JSON_UNESCAPED_SLASHES);
$timestamp = (string)round(microtime(true) * 1000);
$message = $timestamp . $method . $path . $body;
$signature = base64_encode(hash_hmac('sha256', $message, $secretKey, true));
$url = "https://api.ripio.com" . $payload['path'];

// Initialize cURL session
$ch = curl_init($url);

// Set cURL options
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    'Content-Type: application/json',
    'Authorization: ' . $apiKey,
    'Signature: ' . $signature,
    'Timestamp: ' . $timestamp
]);

// Execute cURL session and get the response
$response = curl_exec($ch);

// Check for cURL errors
if (curl_errno($ch)) {
    echo 'cURL error: ' . curl_error($ch);
} else {
    // Decode and print the response
    $result = json_decode($response, true);
    echo "response: ";
    print_r($result);
}

// Close cURL session
curl_close($ch);
